osquery/tables/networking/windows/windows_firewall_rules.cpp

```diff
@@ -0,0 +1,203 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <string>
+
+#include <atlbase.h>
+#include <netfw.h>
+#include <windows.h>
+
+#include <osquery/core/core.h>
+#include <osquery/core/tables.h>
+#include <osquery/logger/logger.h>
+#include <osquery/utils/conversions/windows/strings.h>
+
+namespace osquery {
+namespace tables {
+
+HRESULT populateRow(INetFwRule* rule, Row& r);
+
+QueryData genFirewallRules(QueryContext& context) {
```

Comment:
This function should make sure of the `Expected<Value, Error>` (`utils/expected/expected.h`) class and either return an error or the requested data in the form of a list of structures (without generating rows).